tests/db/migration_tests/test_v202301110_to_v202301120.py

```diff
@@ -0,0 +1,72 @@
+"""
+    Copyright 2023 Inmanta
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+        http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+    Contact: code@inmanta.com
+"""
+import os
+import uuid
+from collections import abc
+from datetime import datetime
+
+import pytest
+from asyncpg import Connection
+
+
+@pytest.mark.db_restore_dump(os.path.join(os.path.dirname(__file__), "dumps/v202301110.sql"))
+async def test_timestamp_timezones(
+    migrate_db_from: abc.Callable[[], abc.Awaitable[None]],
+    postgresql_client: Connection,
+) -> None:
+    """
+    Test that all timestamps are timezone-aware.
+    """
+
+    # Fill the tables with dummy data to be able to fetch the types
```

Comment:
This is what we talked about on Tuesday: we should try to get out of this habit of prepopulating in the test case and instead do so in the dump tool.

If this proves difficult, for a trivial change like this, I think we can get away with leaving it untested. I wouldn't spend too much time on it.

If you really want to test this, it could be more useful to add a generic (non-migration) test that checks that all tables with time columns use timezones, in the same vein as `test_db_schema_enum_consistency`.

Reply:
> This is what we talked about on Tuesday: we should try to get out of this habit of prepopulating in the test case and instead do so in the dump tool.


I don't clearly remember the discussion on Tuesday, prepopulating in the test case was causing some issues ? I don't see what the difference would be to do so in the dump tool.

> If this proves difficult, for a trivial change like this, I think we can get away with leaving it untested. I wouldn't spend too much time on it.

I tend to agree

> If you really want to test this, it could be more useful to add a generic (non-migration) test that checks that all tables with time columns use timezones, in the same vein as test_db_schema_enum_consistency.


I'll look into this :+1:

Reply:
> I don't see what the difference would be to do so in the dump tool.

The main difference is that it then becomes part of the dump, which is static. Then again, I think this approach only starts causing issues if you start using any code from `inmanta` (because that might change to reflect database changes which do not exist yet for this dump, rendering the code incompatible with the pre-state of the database). But you don't do that here, so maybe we should consider it acceptable in this case. @wouterdb wdyt? Should we discuss this special case next week?

Reply:
I'll add some comments. 

https://github.com/inmanta/inmanta-core/pull/5362

Reply:
But the migration script that introduced this table in the first place has never been released, so does it matter?

Reply:
> as we have never used this type of conversion before.

We have: https://github.com/inmanta/inmanta-core/blob/9f8de245b21765d064bc61f14e2397ddf64f602f/src/inmanta/db/versions/v202105170.py#L48